File: sources/Luna b8/lunadevs/luna/module/player/Parkour.java
package lunadevs.luna.module.player;

import lunadevs.luna.category.Category;
import lunadevs.luna.module.Module;

public class Parkour extends Module{

	public Parkour() {
		super("Parkour", 0, Category.PLAYER, false);
	}
	
	@Override
	public void onUpdate() {
		if(!this.isEnabled) return;
	    if ((mc.thePlayer.onGround) && (!mc.thePlayer.isSneaking()) && 
	    	      (!mc.gameSettings.keyBindSneak.pressed)) {
	    	      if (mc.theWorld.getCollidingBoundingBoxes(mc.thePlayer, mc.thePlayer.getEntityBoundingBox().offset(0.0D, -0.5D, 0.0D).expand(-0.001D, 0.0D, -0.001D)).isEmpty()) {
	    	        mc.thePlayer.jump();
	    	      }
	    	    }
		super.onUpdate();
	}

	@Override
	public String getValue() {
		return null;
	}

}

File: sources/Kevin Reborn/kevin/module/modules/movement/Parkour.kt
/*
 * This program is free software: you can redistribute it and/or modify
 *   it under the terms of the GNU General Public License as published by
 *   the Free Software Foundation, either version 3 of the License, or
 *   (at your option) any later version.
 *
 *   This program is distributed in the hope that it will be useful,
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *   GNU General Public License for more details.
 *
 *   You should have received a copy of the GNU General Public License
 *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package kevin.module.modules.movement

import kevin.event.EventTarget
import kevin.event.UpdateEvent
import kevin.module.Module
import kevin.module.ModuleCategory
import kevin.utils.MovementUtils

class Parkour: Module("Parkour", "Automatically jumps when reaching the edge of a block.", category = ModuleCategory.MOVEMENT) {
    @EventTarget
    fun onUpdate(event: UpdateEvent) {
        val thePlayer = mc.thePlayer ?: return

        if (MovementUtils.isMoving && thePlayer.onGround && !thePlayer.isSneaking && !mc.gameSettings.keyBindSneak.isKeyDown && !mc.gameSettings.keyBindJump.isKeyDown &&
            mc.theWorld!!.getCollidingBoundingBoxes(thePlayer, thePlayer.entityBoundingBox
                .offset(0.0, -0.5, 0.0).expand(-0.001, 0.0, -0.001)).isEmpty())
            thePlayer.jump()
    }
}